### PR TITLE
fileoptimizer: Add version 14.10.2534

### LINKED ISSUE #3615: [Request] Add File Optimizer
File Optimizer is an excellent tool for file optimization and minification.
https://nikkhokkho.sourceforge.io/static.php?page=FileOptimizer

### PR DESCRIPTION
**Pull request by issaclin32**
close #3615

~FileOptimizer~ **File Optimizer** ([homepage](https://nikkhokkho.sourceforge.io/static.php?page=FileOptimizer)) is a file optimizer featuring lossless file size reduction.

Notes:
* The *AGPL-3.0-or-later* license can be found in *help -> about* dialog of the main program.

* *FileOptimizer* uses `FileOptimizer32.ini` for 32-bit, and `FileOptimizer64.ini` for 64-bit. ~Therefore I use copy-over method to *persist* the files.~ persist both files instead.

### CODE REVIEW

File: bucket/fileoptimizer.json [hunk 1]
```diff
@@ -0,0 +1,59 @@
...
+    "license": "AGPL-3.0-or-later",
+    "description": "File optimizer featuring lossless file size reduction",
+    "url": "https://downloads.sourceforge.net/project/nikkhokkho/FileOptimizer/14.10.2534/FileOptimizerFull.7z.exe#/dl.7z",
+    "hash": "sha1:a6ba351d9f5cfbd647c64b37f2d4b26c3c2bfd4d",
+    "architecture": {
+        "32bit": {
+            "pre_install": [
+                "if (!(Test-Path \"$persist_dir\\FileOptimizer32.ini\")) { New-Item \"$persist_dir\\FileOptimizer32.ini\" -Force | Out-Null }",
+                "Copy-Item \"$persist_dir\\FileOptimizer32.ini\" \"$dir\\FileOptimizer32.ini\" | Out-Null"
+            ],
```
Comment (Ash258) [inline bucket/fileoptimizer.json]:
```suggestion
```

File: bucket/fileoptimizer.json [hunk 2]
```diff
@@ -0,0 +1,59 @@
...
+            "uninstaller": {
+                "script": "Copy-Item \"$dir\\FileOptimizer32.ini\"  \"$persist_dir\\FileOptimizer32.ini\" -Force | Out-Null"
+            },
```
Comment (Ash258) [inline bucket/fileoptimizer.json]:
```suggestion
```

File: bucket/fileoptimizer.json [hunk 3]
```diff
@@ -0,0 +1,59 @@
...
+            ]
+        },
+        "64bit": {
+            "pre_install": [
+                "if (!(Test-Path \"$persist_dir\\FileOptimizer64.ini\")) { New-Item \"$persist_dir\\FileOptimizer64.ini\" -Force | Out-Null }",
+                "Copy-Item \"$persist_dir\\FileOptimizer64.ini\" \"$dir\\FileOptimizer64.ini\" | Out-Null"
+            ],
+            "uninstaller": {
+                "script": "Copy-Item \"$dir\\FileOptimizer64.ini\"  \"$persist_dir\\FileOptimizer64.ini\" -Force | Out-Null"
+            },
```
Comment (Ash258) [inline bucket/fileoptimizer.json]:
```suggestion
```

File: bucket/fileoptimizer.json [hunk 4]
```diff
@@ -0,0 +1,59 @@
+{
+    "homepage": "https://nikkhokkho.sourceforge.io/static.php?page=FileOptimizer",
+    "version": "14.10.2534",
+    "license": "AGPL-3.0-or-later",
+    "description": "File optimizer featuring lossless file size reduction",
```
Comment (Ash258) [inline bucket/fileoptimizer.json]:
```suggestion
    "version": "14.10.2534",
    "description": "Lossless file size reduction",
    "homepage": "https://nikkhokkho.sourceforge.io/static.php?page=FileOptimizer",
    "license": "AGPL-3.0-or-later",
```

File: bucket/file-optimizer.json [hunk 5]
```diff
@@ -0,0 +1,53 @@
...
+                    "FileOptimizer64.exe",
+                    "File Optimizer"
+                ]
+            ]
+        }
+    },
+    "pre_install": [
+        "if (!(Test-Path \"$persist_dir\\FileOptimizer32.ini\")) { New-Item \"$dir\\FileOptimizer32.ini\" -Force | Out-Null }",
+        "if (!(Test-Path \"$persist_dir\\FileOptimizer64.ini\")) { New-Item \"$dir\\FileOptimizer64.ini\" -Force | Out-Null }"
+    ],
```
Comment (Ash258) [inline bucket/file-optimizer.json]: - ...
- Why force?
     - You need force only if the file already exists, and if it already exists you do not need to do the persist check ...

```
'32', '64' | % { if Test-Path $pers\FileOptimizer$_.ini { New-Item | Out-Null  } }
```

Comment (issaclin32) [reply bucket/file-optimizer.json]: > * ...
> * Why force?
>   * You need force only if the file already exists, and if it already exists you do not need to do the persist check ...
It's my fault. `Force` is only required in copy-over method. I didn't edit it out.

File: bucket/file-optimizer.json [hunk 6]
```diff
@@ -0,0 +1,50 @@
...
+            ],
+            "shortcuts": [
+                [
+                    "FileOptimizer64.exe",
+                    "File Optimizer"
+                ]
+            ]
+        }
+    },
+    "pre_install": "'32', '64' | % { if (!(Test-Path $persist_dir\\FileOptimizer$_.ini)) { New-Item | Out-Null } }",
```
Comment (Ash258) [inline bucket/file-optimizer.json]: Are you kidding me?

I hope that need for editing it as normal, readable and working PowerShell is for someone who wrote so much manifest do not have to be mentioned...

Comment (issaclin32) [reply bucket/file-optimizer.json]: Now it should be working properly. Really sorry for that.

File: bucket/file-optimizer.json [hunk 7]
```diff
@@ -0,0 +1,50 @@
...
+            ],
+            "shortcuts": [
+                [
+                    "FileOptimizer64.exe",
+                    "File Optimizer"
+                ]
+            ]
+        }
+    },
+    "pre_install": "'32', '64' | % { if (!(Test-Path \"$persist_dir\\FileOptimizer$_.ini\")) { New-Item \"$dir\\FileOptimizer$_.ini\" | Out-Null } }",
```
Comment (Ash258) [inline bucket/file-optimizer.json]:
```suggestion
    "pre_install": [
         "'32', '64' | ForEach-Object {",
         "    if (!(Test-Path \"$persist_dir\\FileOptimizer$_.ini\")) { New-Item \"$dir\\FileOptimizer$_.ini\" | Out-Null }",
         "}"
     ]
```

**Comment by Ash258** [comment]
Author intention with name is not much clear. I changed it to `fileoptimizer` as this name is in newer version everywhere

**Pull request merged**